compiler/rustc_mir_transform/src/coverage/mod.rs

```diff
@@ -96,10 +96,17 @@ fn instrument_function_for_coverage<'tcx>(tcx: TyCtxt<'tcx>, mir_body: &mut mir:
     }
 
     let bcb_has_counter_mappings = |bcb| bcbs_with_counter_mappings.contains(bcb);
-    let coverage_counters =
+    let mut coverage_counters =
         CoverageCounters::make_bcb_counters(&basic_coverage_blocks, bcb_has_counter_mappings);
 
-    let mappings = create_mappings(tcx, &hir_info, &extracted_mappings, &coverage_counters);
+    let mappings = create_mappings(
+        mir_body,
+        tcx,
+        &hir_info,
+        &mut extracted_mappings,
+        &basic_coverage_blocks,
+        &mut coverage_counters,
```

Comment:
Then some keys may be essential to mappings so that we can get these sum expressions later, sounds a bit redundant.
Currently we only need to create `Expression` but not `Counter` here. What are the drawbacks of the current approach?

Reply:
After some more thought, I think it's OK to do this for now.

I still have mixed feelings about it, but I don't know what the specific alternative should look like, and I don't think this change should have to wait for me to figure that out.